pkg/encoding/base58/base58.go

```diff
@@ -16,13 +16,6 @@ func CheckDecode(s string) (b []byte, err error) {
 		return nil, err
 	}
 
-	for i := 0; i < len(s); i++ {
```

Comment:
This is needed when base58 is used to encode a number where `001 == 1`. `1` are used to encode leading `0`.
`mr-tron`'s implementation does this by default, so you have actually fixed a bug.

This test fails on master, can you add it?
```golang
func TestBase58LeadingZeroes(t *testing.T) {
	buf := []byte{0, 0, 0, 1}
	b58 := CheckEncode(buf)
	dec, err := CheckDecode(b58)
	require.NoError(t, err)
	require.Equal(t, buf, dec)
}
```

Reply:
Thanks for explanation, this must be a remnant of internal base58 implementation then. Added a test.